lib/widgets/radio_emoji.dart

```diff
@@ -0,0 +1,172 @@
+              const Icon(
+                Icons.sentiment_satisfied,
+                color: Colors.lightGreen,
+                size: 30,
+              ),
+              const Icon(
+                Icons.sentiment_very_satisfied,
+                color: Colors.green,
+                size: 30,
+              )
+            ];
+            final List<SelectableController> emojiControllers =
+                emojis.map((_) => SelectableController()).toList();
+
+            final emojiSelectables = emojiControllers
+                .asMap()
+                .map(
+                  (i, controller) => MapEntry(
+                    i,
+                    SelectableIcon(
+                      icon: emojis[i],
+                      controller: controller,
+                      onSelected: (selected) {
+                        if (selected) {
+                          for (final c in emojiControllers) {
+                            if (c != controller) {
+                              c.deselect();
+                              state.value[i] = selected;
+                              state.value[emojiControllers.indexOf(c)] =
+                                  !selected;
+                            }
+                          }
+                          controller.select();
+                          state.value[i] = selected;
+                          state.didChange(state.value);
+                        } else {
+                          controller.deselect();
+                          state.value[i] = selected;
+                          state.didChange(state.value);
+                        }
+                      },
+                    ),
+                  ),
+                )
+                .values
+                .toList();
+            return Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: [
+                const SizedBox(height: 24),
+                Text(
+                  '$question',
+                  style: const TextStyle(
+                    fontSize: 18,
+                  ),
+                ),
+                Container(
+                  width: MediaQuery.of(context).size.width - 20,
+                  child: Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: emojiSelectables,
+                  ),
+                ),
+                const SizedBox(height: 12),
+                if (state.hasError)
+                  Padding(
+                    padding: const EdgeInsets.only(top: 8),
+                    child: Text(
+                      state.errorText,
+                      style: Theme.of(context)
+                          .textTheme
+                          .caption
+                          .copyWith(color: Theme.of(context).errorColor),
+                    ),
+                  ),
+              ],
+            );
+          },
+        );
+}
+
+class SelectableIcon extends Selectable {
```

Comment:
Try to not use selectable anymore so we can delete it. Instead of "initiallySelected", make it a non-final variable called "selected" and try to just modify it in the state, I think it should work to see it outside of the widget as well. If you can't get this to work, don't spend more than 1hr on it, PM me and I'll try to help.